framework/src/media/Encoder.cpp

```diff
@@ -0,0 +1,130 @@
+/* ****************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+#include <tinyara/config.h>
+#include <string.h>
+#include "Encoder.h"
+#include <debug.h>
+
+#define PV_SUCCESS 0
+#define PV_FAILURE -1
+
+namespace media {
+
+Encoder::Encoder(audio_type_t audio_type)
+{
+#ifdef CONFIG_AUDIO_CODEC
+	switch (audio_type) {
+
+#ifdef CONFIG_CODEC_LIBOPUS
+	case AUDIO_TYPE_OPUS: {
+		memset(&mEncoder, 0, sizeof(aud_encoder_t));
+
+		static uint8_t outputBuf[1024];
+		static int16_t inputBuf[1920*2];  // inputSampleRate / 1000 * frame_size_ms * 2
+
+		opus_enc_external_t ext = {0};
+		// params for opus encoding
+		ext.applicationMode = OPUS_APPLICATION_AUDIO;
+		ext.frameSizeMS = 100;
+		ext.bitrate = 16000;
+		ext.bandWidth = OPUS_AUTO;
+		ext.complexity = 5;
```

Comment:
I think mode & complexity is one of main features of opus, why don't you define them in KConfig or somewhere instead of using constant value?